# 1. 라이브러리 불러오기 및 데이터셋 path설정

In [280]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
import seaborn as sns

path = 'dataset/'

# 2. 데이터 불러오기 및 정렬

In [327]:
row_data = pd.read_csv('{}v7_final.csv'.format(path), encoding='euc-kr') #, index_col='Unnamed: 0')
row_data.head()

회사명  거래소코드     회계년도     산업코드  소속코드         상장일 상장폐지일 감사의견코드  총자본증가율  \
0  (주)CMG제약  58820  2007/12  32102.0   5.0  2001/08/31   NaN     UQ  120.08   
1  (주)CMG제약  58820  2008/12  32102.0   5.0  2001/08/31   NaN     UQ   56.55   
2  (주)CMG제약  58820  2009/12  32102.0   5.0  2001/08/31   NaN     UQ  -33.92   
3  (주)CMG제약  58820  2010/12  32102.0   5.0  2001/08/31   NaN     UQ    5.66   
4  (주)CMG제약  58820  2011/12  32102.0   5.0  2001/08/31   NaN     UQ   14.18   

   유형자산증가율  ...  PCR(Price cash-flow ratio)(최고)  \
0   215.25  ...                             0.0   
1    55.67  ...                             0.0   
2    -8.50  ...                             0.0   
3   -11.92  ...                             0.0   
4     7.86  ...                             0.0   

   PCR(Price cash-flow ratio)(최저)  PSR(Price sales ratio)(최고)  \
0                             0.0                      149.75   
1                             0.0                        9.13   
2                             0.0                        8.40   
3                             0.0                        3.22   
4                             0.0                        3.14   

   PSR(Price sales ratio)(최저)  기업가치(EV)(백만원)  EBITDA(백만원)  EBITDA/매출액(%)  \
0                       20.92       69545.33     -1288.45         -71.74   
1                        1.49        -454.50     -3419.79         -31.64   
2                        1.72       26517.58     -7100.69         -69.98   
3                        1.04       31200.74     -5598.06         -49.32   
4                        1.05       37758.74      2038.64          11.48   

   EBITDA/금융비용(배)  EBITDA/평균발행주식수(백만원)  EV/EBITDA(배)  
0            0.00              -104.44          0.00  
1           -4.56              -221.19           NaN  
2           -6.80              -443.98           NaN  
3           -5.51              -243.34           NaN  
4           11.24                46.75         18.52  

[5 rows x 174 columns]

In [328]:
row_data = row_data[~row_data['산업코드'].isnull()]

In [329]:
row_data['소속코드'] = row_data['소속코드'].astype(int)
row_data['산업코드'] = row_data['산업코드'].astype(int)

In [330]:
row_data['소속코드'] = row_data['소속코드'].astype(str).str.zfill(2)
row_data['산업코드'] = row_data['산업코드'].astype(str).str.zfill(6)
row_data['거래소코드'] = row_data['거래소코드'].astype(str).str.zfill(6)

In [331]:
row_data['산업대분류'] = row_data['산업코드'].str[0:2]

In [332]:
# 원데이터에서 에초에 모집단으로 선정한 산업대분류 코드 3, 10, 7을 추출하여 새로운 data를 만들어줍니다.
data = row_data[(row_data['산업대분류'] == '03') |
                (row_data['산업대분류'] == '10') | 
                (row_data['산업대분류'] == '07')]

data['회계년도_new']=data['회계년도'].str[0:4].astype(int)
data['상장일_new']=data['상장일'].str[0:4].astype(int)

/var/folders/1v/ctxpzb994v98h43xyk972tmc0000gn/T/ipykernel_6332/845699834.py:6: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['회계년도_new']=data['회계년도'].str[0:4].astype(int)
/var/folders/1v/ctxpzb994v98h43xyk972tmc0000gn/T/ipykernel_6332/845699834.py:7: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['상장일_new']=data['상장일'].str[0:4].astype(int)


In [333]:
# 부도여부 칼럼 형성
data.loc[((data['감사의견코드'] == 'DS') | 
          (data['감사의견코드'] == 'DU')) & 
         (data['소속코드'].isin(['03','04','06'])), 
         '부도여부'] = '1'

data['부도여부'] = data['부도여부'].fillna('0')

/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pandas/core/indexing.py:1684: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self.obj[key] = infer_fill_value(value)
/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pandas/core/indexing.py:1817: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._setitem_single_column(loc, value, pi)
/var/folders/1v/ctxpzb994v98h43xyk972tmc0000gn/T/ipykernel_6332/3089269643.py:7: SettingWithCopyWarning: 
A value is try

In [334]:
# 초기 데이터 셋에서 부도 기업의 행 수
data['부도여부'].value_counts()

0    20934
1      249
Name: 부도여부, dtype: int64

## 3. 전처리 시행
### 3-1. 회사명이 같은데 거래소코드가 다른 경우 (워드파일 4-A)

- check를 통해서 기업 리스트를 뽑아봄
- check_df를 통해서 check 기업의 데이터를 확인해봄

In [335]:
check = data.groupby(['회사명'])['거래소코드'].nunique().reset_index()
check.rename(columns={'거래소코드':'거래소코드_갯수'}, inplace=True)
check = check[ check['거래소코드_갯수']>1]
# check

In [336]:
check_df = data[data['회사명'].isin(check['회사명'].tolist())]
check_df = check_df.reset_index()
# check_df

In [337]:
# 회사명이 같은데 거래소코드가 다른 경우를 대상으로 행 개수를 출력해봄
check_df.groupby(['회사명','거래소코드'])['index'].count().reset_index()

회사명   거래소코드  index
0         (주)본느  226340      6
1         (주)본느  242420      2
2      (주)세화피앤씨  085980      4
3      (주)세화피앤씨  252500      5
4      (주)신성이엔지  011930     14
5      (주)신성이엔지  104110      8
6    (주)아이엘사이언스  122050      1
7    (주)아이엘사이언스  307180      3
8       (주)알로이스  271400      2
9       (주)알로이스  297570      3
10     (주)엔케이맥스  182400      6
11     (주)엔케이맥스  262760      1
12       (주)인산가  062580      3
13       (주)인산가  277410      4
14      (주)인터파크  035080     14
15      (주)인터파크  108790      6
16       (주)자비스  230400      4
17       (주)자비스  254120      5
18       (주)정다운  208140      7
19       (주)정다운  224090      1
20   (주)티에스트릴리온  284610      3
21   (주)티에스트릴리온  317240      2
22  (주)포인트엔지니어링  176560      2
23  (주)포인트엔지니어링  256630      4
24       (주)풀무원  017810     14
25       (주)풀무원  103160      1
26     (주)한국비엔씨  226610      4
27     (주)한국비엔씨  256840      5
28     (주)핸디소프트  032380      3
29     (주)핸디소프트  220180      5
30      나무기술(주)  167380      2
31      나무기술(주)  242040      5
32      삼성물산(주)  000830      8
33      삼성물산(주)  028260      7
34     소프트캠프(주)  210610      5
35     소프트캠프(주)  258790      4
36     씨아이에스(주)  144630      2
37     씨아이에스(주)  222080      6
38      에스와이(주)  004530      1
39      에스와이(주)  109610      6
40     엔에이치엔(주)  035420      1
41     엔에이치엔(주)  181710      8
42   엠에프엠코리아(주)  251960      4
43   엠에프엠코리아(주)  323230      2
44      줌인터넷(주)  229480      3
45      줌인터넷(주)  239340      5
46     케이엠제약(주)  225430      6
47     케이엠제약(주)  237720      3
48     패션플랫폼(주)  225590      6
49     패션플랫폼(주)  256090      1

In [338]:
# 합병등과 같은 이유로 거래소코드가 2개인 경우 데이터프레임에서 삭제
data = data[~data['회사명'].isin(check_df['회사명'])]

In [339]:
data['부도여부'].value_counts()

0    20708
1      248
Name: 부도여부, dtype: int64

#### 3-1. 결론 : 버린다.
- 예를들어 7월에 합병되는 경우, 피인수기업 데이터는 1/1~07/01의 데이터를 기반으로 재무비율을 형성하고 인수기업의 경우에는 피인수기업의 재무적 정보를 그대로 가져오는 등의 경우의 수가 다양하기 때문에, 이러한 데이터를 drop하는 것으로 생각한다.
- 또한, 해당 기업 25개 중 부도로 집계된 기업은 1개에 불과하다. 따라서 정상기업이 많고 약 2250개에서 25개는 1%이므로 이는 제거하더라도 모델의 성능에 영향을 덜 미칠 것으로 판단된다. 따라서 제거한다.) 
- 위의 부도여부가 1인 값은 앞과 비교했을 때, 248개로 1개만 감소한 것을 확인할 수 있다.

### 3-2 부도, 정상 기업 데이터 정리 : 워드(4- B)
- 회계년도와 거래소 코드를 기준으로 전년도 재무비율과 당해년도 부도여부로 구성된 데이터 셋을 형성해야 함.

In [364]:
data['left_on'] = data['회계년도_new']-1
data['회계년도_new'].astype(int)

left_df = data[['회사명', '거래소코드','부도여부','left_on']]

In [370]:
merged_df = pd.merge(left_df, data, left_on=['회사명','거래소코드','left_on'], 
                     right_on = ['회사명','거래소코드','회계년도_new'], how='left')
merged_df.head()
# 여기에 Nan값이 나오는 이유
# left_on_x의 처음 값이 2006년인데, 우리는 데이터셋을 에초에 2007년부터 불러왔기 때문에 없는 것임.
# 따라서 해당 행을 drop해야 한다.

회사명   거래소코드 부도여부_x  left_on_x     회계년도    산업코드 소속코드         상장일 상장폐지일  \
0  (주)CMG제약  058820      0       2006      NaN     NaN  NaN         NaN   NaN   
1  (주)CMG제약  058820      0       2007  2007/12  032102   05  2001/08/31   NaN   
2  (주)CMG제약  058820      0       2008  2008/12  032102   05  2001/08/31   NaN   
3  (주)CMG제약  058820      0       2009  2009/12  032102   05  2001/08/31   NaN   
4  (주)CMG제약  058820      0       2010  2010/12  032102   05  2001/08/31   NaN   

  감사의견코드  ...  EBITDA(백만원)  EBITDA/매출액(%)  EBITDA/금융비용(배)  \
0    NaN  ...          NaN            NaN             NaN   
1     UQ  ...     -1288.45         -71.74            0.00   
2     UQ  ...     -3419.79         -31.64           -4.56   
3     UQ  ...     -7100.69         -69.98           -6.80   
4     UQ  ...     -5598.06         -49.32           -5.51   

   EBITDA/평균발행주식수(백만원)  EV/EBITDA(배)  산업대분류  회계년도_new  상장일_new  부도여부_y  \
0                  NaN           NaN    NaN       NaN      NaN     NaN   
1              -104.44           0.0     03    2007.0   2001.0       0   
2              -221.19           NaN     03    2008.0   2001.0       0   
3              -443.98           NaN     03    2009.0   2001.0       0   
4              -243.34           NaN     03    2010.0   2001.0       0   

   left_on_y  
0        NaN  
1     2006.0  
2     2007.0  
3     2008.0  
4     2009.0  

[5 rows x 181 columns]

In [371]:
# 2006년 데이터는 애초에 없기에 그런 행들을 삭제해준다.
merged_df = merged_df[~merged_df['회계년도'].isnull()]

merged_df.drop(columns = ['부도여부_y','left_on_x','left_on_y'], inplace=True)
merged_df.rename(columns = {'부도여부_x': '차기부도여부'}, inplace=True)
merged_df.shape

# 18723 행

(18723, 178)

In [351]:
print('부도 기업의 데이터 행 수 : {}'.format(merged_df['차기부도여부'].value_counts()[1]))
print('다음 사업연도에 감삭결과가 DS, DU 이거나 관리종목 혹은 상장 폐지로 되는 기업의 개수 : {}'.format(merged_df[merged_df['차기부도여부']=='1']['회사명'].nunique()))

부도 기업의 데이터 행 수 : 234
다음 사업연도에 감삭결과가 DS, DU 이거나 관리종목 혹은 상장 폐지로 되는 기업의 개수 : 214


In [354]:
# 부도_df는 위의 부도기업 데이터 행 수 234개에 해당하는 데이터를 뽑은 것이다. (중복이 20개 존재함)
qneh_df = merged_df[merged_df['차기부도여부']=='1']
qneh_df = qneh_df.reset_index()

# 그렇다면, 부도_df를 거래소코드로 묶은 뒤 칼럼의 수를 구하고 컬럼의 이름을 부도_cnt로 설정해서 뽑을 수 있다.
check = qneh_df.groupby(['거래소코드'])['index'].count().reset_index()
check.rename(columns={'index':'부도_cnt'}, inplace=True)

In [363]:
# 이 부분은 부도 기업의 행 수가 2개 이상인 기업을 뽑는 부분임.
# check[check['부도_cnt']>1]

# 여기서 거래소코드를 리스트화하여 뽑을 수 있음.
company = check[check['부도_cnt'] > 1]['거래소코드'].tolist()

In [360]:
merged_df[merged_df['거래소코드'].isin(company)]

회사명   거래소코드 차기부도여부     회계년도    산업코드 소속코드         상장일       상장폐지일  \
2553     (주)모다  149940      1  2018/12  032604   04  2012/09/25  2020/07/10   
2554     (주)모다  149940      1  2016/12  032604   04  2012/09/25  2020/07/10   
2555     (주)모다  149940      1  2017/12  032604   04  2012/09/25  2020/07/10   
2556     (주)모다  149940      0  2012/12  032604   04  2012/09/25  2020/07/10   
2557     (주)모다  149940      0  2014/12  032604   04  2012/09/25  2020/07/10   
...        ...     ...    ...      ...     ...  ...         ...         ...   
20796  현진소재(주)  053660      0  2012/12  032509   06  2002/02/15         NaN   
20797  현진소재(주)  053660      0  2008/12  032509   06  2002/02/15         NaN   
20798  현진소재(주)  053660      0  2010/12  032509   06  2002/02/15         NaN   
20800  현진소재(주)  053660      0  2011/12  032509   06  2002/02/15         NaN   
20801  현진소재(주)  053660      0  2013/12  032509   06  2002/02/15         NaN   

      감사의견코드  총자본증가율  ...  PSR(Price sales ratio)(최저)  기업가치(EV)(백만원)  \
2553      DU  -71.32  ...                        0.16       38948.62   
2554      UQ  271.12  ...                        1.16      180931.13   
2555      DS   77.20  ...                        3.18      271730.00   
2556      UQ   17.27  ...                        0.79       28712.18   
2557      UQ   10.97  ...                        2.07       64869.09   
...      ...     ...  ...                         ...            ...   
20796     UQ  -10.54  ...                        0.28      257278.15   
20797     UQ   73.03  ...                        0.39       59702.56   
20798     UQ   -5.11  ...                        0.77      376174.73   
20800     UQ   14.82  ...                        0.35      308750.84   
20801     UQ    5.74  ...                        0.25      266260.70   

       EBITDA(백만원)  EBITDA/매출액(%)  EBITDA/금융비용(배)  EBITDA/평균발행주식수(백만원)  \
2553      -9948.29         -51.15           -1.33              -470.14   
2554       5849.42          12.43            1.78               492.36   
2555     -19020.77         -63.57           -1.49             -1004.34   
2556       2650.84           7.10           60.09               552.10   
2557      -4998.73         -25.94          -10.64              -509.24   
...            ...            ...             ...                  ...   
20796     18848.35           5.53            2.08              1273.54   
20797     91837.60          23.01           20.19              6444.74   
20798     -9919.51          -3.52           -1.88              -681.75   
20800     26267.90           6.48            2.93              1774.86   
20801      1579.87           0.68            0.19               106.75   

       EV/EBITDA(배)  산업대분류  회계년도_new  상장일_new  
2553           0.00     03    2018.0   2012.0  
2554          30.93     03    2016.0   2012.0  
2555           0.00     03    2017.0   2012.0  
2556          10.83     03    2012.0   2012.0  
2557           0.00     03    2014.0   2012.0  
...             ...    ...       ...      ...  
20796         13.65     03    2012.0   2002.0  
20797          0.65     03    2008.0   2002.0  
20798          0.00     03    2010.0   2002.0  
20800         11.75     03    2011.0   2002.0  
20801        168.53     03    2013.0   2002.0  

[171 rows x 178 columns]

## 2021.11.29 작업 종료

In [257]:
# data.sort_values(by=['회사명','회계년도'], inplace=True)

/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pandas/util/_decorators.py:311: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return func(*args, **kwargs)


In [252]:
# data.loc[data['부도여부'] == '1']['회사명'].nunique() #.value_counts()
# data.loc[data['부도여부'] == '1']['거래소코드'].nunique()

In [258]:
# data['sum_nan'] =data[data['상장일_new'] == data['회계년도_new']].isnull().sum(1)

data['sum_nan'] = data.isnull().sum(1)

/var/folders/1v/ctxpzb994v98h43xyk972tmc0000gn/T/ipykernel_6332/562235871.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['sum_nan'] = data.isnull().sum(1)


In [259]:
# data['sum_nan'] = data['sum_nan'].fillna(0)
# data = data.loc[data['sum_nan'] <167]
# data.drop(columns='sum_nan', inplace= True)
data['sum_nan'].value_counts()

2      11997
11      6697
1       1166
10       547
3        361
168      263
0         81
9         41
169       15
167       11
17         2
38         1
30         1
Name: sum_nan, dtype: int64

In [ ]:
check_df = data[data['회사명'].isin(check['회사명'].tolist())]
check_df = check_df.reset_index()
# check_df

In [188]:
data['left_on'] = data['회계년도_new']-1

/var/folders/1v/ctxpzb994v98h43xyk972tmc0000gn/T/ipykernel_6332/2157095088.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['left_on'] = data['회계년도_new']-1


In [83]:
# 재무비율 명칭을 정렬합니다.
financial_ratio = []
for i in data.columns[11:-1]:
    financial_ratio.append(i)
    print(i)

len(financial_ratio)

총자본증가율
유형자산증가율
유동자산증가율
영업이익증가율
경상이익증가율 (2007년 이전 발생)
순이익증가율
재고자산증가율
자기자본증가율
매출액증가율
종업원1인당 부가가치증가율
종업원수증가율
비유동자산증가율
종업원1인당 매출액증가율
종업원1인당 인건비증가율
매출액총이익률
매출액영업이익률
매출액경상이익률 (2007년 이전 발생)
매출액순이익률
총자본사업이익률
총자본영업이익률
총자본경상이익률 (2007년 이전 발생)
총자본순이익률
자기자본영업이익률
자기자본경상이익률 (2007년 이전 발생)
자기자본순이익률
자본금영업이익률
자본금경상이익률 (2007년 이전 발생)
자본금순이익률
조세 대 조세차감전순이익률
기업경상이익률 (2007년 이전 발생)
투자수익률 (2007년 이전 발생)
기업순이익률
경영자본영업이익률
경영자본순이익률
매출원가 대 매출액비율
영업비율
영업외손익률
금융비용부담률
외환이익 대 매출액비율
광고선전비 대 매출액비율
수지비율
인건비 대 총비용비율
조세공과 대 총비용비율
금융비용 대 총비용비율
감가상각비 대 총비용비율
감가상각률
누적감가상각률
이자부담률
지급이자율
차입금평균이자율
사내유보율
사내유보 대 자기자본비율
적립금비율(재정비율)
평균배당률
자기자본배당률
배당성향
1주당매출액(원)
EPS(Earning Per Share)(원)
1주당경상이익 (2007년 이전 발생)(원)
CPS(Cash flow Per Share)(원)
BPS(Book-value Per Share)(원)
유보율
R & D 투자효율
1주당영업이익(원)
유동자산구성비율
재고자산 대 유동자산비율
유동자산 대 비유동자산비율
당좌자산구성비율
비유동자산구성비율
자기자본구성비율
타인자본구성비율
자기자본배율
비유동비율
비유동장기적합률
유동비율
당좌비율
현금비율
원재료비율
매출채권비율
재고자산 대 순운전자본비율
매출채권 대 매입채무비율
매출채권 대 상
매입채무 대 재고자산비율
부채비율
유동부채비율
단기차입금 대 총차입금비율
비유동부채비율
비유동부채 대 순운전자본비율
순운전자본비율
차입금의존도
차입금비

166

In [39]:
# 기존에 부도기업으로 분류한 dataset을 불러옵니다.
label = pd.read_csv('{}unique_firm.csv'.format(path), encoding='utf-8', index_col='Unnamed: 0')
label.head()

회사명  거래소코드       회계년도    산업코드  업종코드  소속코드         상장일      상장폐지일  \
0  (주)바로크가구  17120  2000.12.1   33201     1     4  1989.12.18  2001.7.10   
1  쌍용자동차(주)   3620  2000.12.1   33001     1     3   1975.5.21        NaN   
2  (주)골드앤에스  35290  2000.12.1  105801     1     6   1999.12.7        NaN   
3     (주)대우   3810  2000.12.1   74607     1     4   1977.6.30  2001.5.23   
4   (주)태화쇼핑  15880  2000.12.1   74701     1     4   1994.11.5  2001.9.22   

  감사의견코드                                               감사의견  산업대분류  회계년도_new  
0     DU  경상손실 23,819백만원 발생, 총자산을 초과한 총부채액이 132,195백만원 발...      3      2000  
1     DU  영업손실 259,221백만원 총자산을 초과한 총부채액이 949,281백만원이 발생....      3      2000  
2     DU  경상손실 8,098백만원 이상, 유동자산을 초과하는 유동부채가 6,657백만원 이상...     10      2000  
3     DU  총부채가 총자산보다 14,031,384백만원 초과,누적결손금이 14,917,023백...      7      2000  
4     DU  경상순손실 22,543백만원 상거래 정리채권 5,996백만원 이자 4,756백만원 미상환      7      2000

In [261]:


label[label['회계년도_new'] == 2020] #['회계년도_new'].value_counts()

회사명   거래소코드       회계년도    산업코드  업종코드  소속코드         상장일 상장폐지일  \
281      (주)매직마이크로  127160  2020.12.1   32602     1     6  2015.11.19   NaN   
282         테라셈(주)  182690  2020.12.1   32601     1     6  2014.10.30   NaN   
283         (주)코스온   69110  2020.12.1   32004     1     6  2003.10.14   NaN   
284       파나케이아(주)   58530  2020.12.1   74604     1     6    2002.2.5   NaN   
285  (주)온코퀘스트파마슈티컬   78590  2020.12.1   33003     1     6  2005.10.27   NaN   
286      명성티엔에스(주)  257370  2020.12.1   32902     1     6   2018.9.13   NaN   
287      (주)에코마이스터   64510  2020.12.1   32902     1     6   2018.3.15   NaN   
288       (주)좋은사람들   33340  2020.12.1   31401     1     6   1997.11.3   NaN   
289    (주)엔지스테크널러지  208860  2020.12.1  105802     1     6   2016.7.28   NaN   
290         (주)뉴로스  126870  2020.12.1   32901     1     6   2012.2.15   NaN   
291   (주)아이티엑스에이아이   99520  2020.12.1   32605     1     6  2009.12.22   NaN   
292       (주)씨엔플러스  115530  2020.12.1   32602     1     6   2011.11.2   NaN   
293          (주)금빛   45890  2020.12.1   32804     1     6  2005.11.28   NaN   
294        (주)제이웨이   58420  2020.12.1  105901     1     6   2002.1.10   NaN   
295     (주)센트럴인사이트   12600  2020.12.1   32901     1     3   1990.7.10   NaN   
296        (주)소리바다   53110  2020.12.1  106301     1     6   2001.10.9   NaN   
297     (주)씨오더블유오엔   56000  2020.12.1  105802     1     6   2003.7.11   NaN   
298         (주)유테크  178780  2020.12.1   74608     1     6   2015.6.30   NaN   
299          (주)성안   11300  2020.12.1   31302     1     3  1995.11.10   NaN   

    감사의견코드 감사의견  산업대분류  회계년도_new  
281     DU  NaN      3      2020  
282     DS  NaN      3      2020  
283     DS  NaN      3      2020  
284     DU  NaN      7      2020  
285     DU  NaN      3      2020  
286     DS  NaN      3      2020  
287     DU  NaN      3      2020  
288     DS  NaN      3      2020  
289     DS  NaN     10      2020  
290     DU  NaN      3      2020  
291     DS  NaN      3      2020  
292     DS  NaN      3      2020  
293     DU  NaN      3      2020  
294     DU  NaN     10      2020  
295     DS  NaN      3      2020  
296     DS  NaN     10      2020  
297     DS  NaN     10      2020  
298     DS  NaN      7      2020  
299     DS  NaN      3      2020

In [267]:
label.loc[label['회사명'] =='(주)인터파크']

Empty DataFrame
Columns: [회사명, 거래소코드, 회계년도, 산업코드, 업종코드, 소속코드, 상장일, 상장폐지일, 감사의견코드, 감사의견, 산업대분류, 회계년도_new]
Index: []

In [268]:
data.loc[data['회사명'] == '(주)인터파크']

회사명   거래소코드     회계년도    산업코드 소속코드         상장일       상장폐지일 감사의견코드  \
10301  (주)인터파크  035080  2007/12  074709   05  1999/07/05         NaN     UQ   
10285  (주)인터파크  035080  2008/12  074709   05  1999/07/05         NaN     UQ   
10286  (주)인터파크  035080  2009/12  074709   05  1999/07/05         NaN     UQ   
10288  (주)인터파크  035080  2010/12  074709   05  1999/07/05         NaN     UQ   
10289  (주)인터파크  035080  2011/12  074709   05  1999/07/05         NaN     UQ   
10290  (주)인터파크  035080  2012/12  074709   05  1999/07/05         NaN     UQ   
10291  (주)인터파크  035080  2013/12  074709   05  1999/07/05         NaN     UQ   
10292  (주)인터파크  035080  2014/12  074709   05  1999/07/05         NaN     UQ   
10293  (주)인터파크  108790  2014/12  074709   04  2014/02/06  2020/07/16     UQ   
10294  (주)인터파크  035080  2015/12  074709   05  1999/07/05         NaN     UQ   
10296  (주)인터파크  108790  2015/12  074709   04  2014/02/06  2020/07/16     UQ   
10295  (주)인터파크  035080  2016/12  074709   05  1999/07/05         NaN     UQ   
10302  (주)인터파크  108790  2016/12  074709   04  2014/02/06  2020/07/16     UQ   
10297  (주)인터파크  108790  2017/12  074709   04  2014/02/06  2020/07/16     UQ   
10298  (주)인터파크  035080  2017/12  074709   05  1999/07/05         NaN     UQ   
10299  (주)인터파크  035080  2018/12  074709   05  1999/07/05         NaN     UQ   
10300  (주)인터파크  108790  2018/12  074709   04  2014/02/06  2020/07/16     UQ   
10284  (주)인터파크  108790  2019/12  074709   04  2014/02/06  2020/07/16     UQ   
10287  (주)인터파크  035080  2019/12  074709   05  1999/07/05         NaN     UQ   
10283  (주)인터파크  035080  2020/12  074709   05  1999/07/05         NaN     UQ   

       총자본증가율  유형자산증가율  ...  EBITDA(백만원)  EBITDA/매출액(%)  EBITDA/금융비용(배)  \
10301   -0.93    -5.37  ...      3389.36          10.45            1.24   
10285   13.06    -5.44  ...     13931.42          41.98            4.67   
10286  109.28   -10.72  ...    -19055.15        -190.60          -16.56   
10288   -8.73     5.04  ...    -14314.74        -165.11            0.00   
10289   46.05   -94.80  ...     -7586.92         -79.85          -35.94   
10290   -7.58   -48.28  ...      -322.12          -2.89           -0.06   
10291   -1.85   -39.18  ...       537.34           6.77            0.29   
10292   -3.68    36.55  ...       403.88           5.21            0.89   
10293   15.86     5.13  ...     21076.23           5.63          130.02   
10294    4.37   -15.66  ...      6029.71          46.85           35.37   
10296   12.97    -4.76  ...     30350.27           8.02          875.63   
10295  -14.97   -10.76  ...     10856.04          65.73           22.81   
10302   -1.23    -2.94  ...     13675.37           3.22        20688.90   
10297   10.68     0.51  ...     20792.15           4.90            0.00   
10298  -14.97    16.42  ...      6143.77          49.21           26.63   
10299   -1.33   -37.86  ...      6065.66          49.56           27.91   
10300    6.34     9.91  ...     11527.93           2.63         1698.28   
10284   -6.33    68.53  ...     26945.94           6.14           60.43   
10287   -0.98  1044.37  ...      4509.46          51.72           13.19   
10283   63.16  2787.44  ...      3954.26           2.34           12.75   

       EBITDA/평균발행주식수(백만원)  EV/EBITDA(배)  산업대분류  회계년도_new  상장일_new  부도여부  \
10301                66.24        108.55     07      2007     1999   NaN   
10285               252.56          2.20     07      2008     1999   NaN   
10286              -313.91           NaN     07      2009     1999   NaN   
10288              -235.61           NaN     07      2010     1999   NaN   
10289              -124.74          0.00     07      2011     1999   NaN   
10290                -5.29          0.00     07      2012     1999   NaN   
10291                 8.83       1280.88     07      2013     1999   NaN   
10292                 6.64       1384.54     07      2014     1999   NaN   
10293               656.96         33.04     07      2014     2014   NaN   
10294          

# 해야할 일
1. train, validation, test 3가지로 기간을 분리해야 합니다.
2. 부도기업 리스트를 활용하여 data를 분류하는 작업을 수행해야 합니다.
3. data를 분류하게 된다면 모집단을 2개로 나눈 후에 T-test를 시행합니다. (financial_ratio를 활용하여 진행)
   1. 하지만, t-test에 대한 가정 (모집단의 정규분포성 등)이 충족되는지 확인해야 합니다.

In [126]:
year_list = list(filter(lambda x: x, range(2000,2021,1)))